convert_representation.py

```diff
@@ -8,23 +12,158 @@ def convert_parenthesized_to_tree(parenthesized: str) -> Graph:
     
     Args:
         parenthesized (str): Reprezentacja nawiasowa struktury RNA.
-    ValueError: Kiedy nie da sie przekonwertowac, bo struktura jest zla
-    
+    Raises:
+        ValueError: Kiedy nie da sie przekonwertowac, bo struktura nawiasów jest zla
     Returns:
-        Graph: Graf reprezentujący strukturę RNA.
+        Graph: Graf reprezentujący strukturę RNA. 
+        Struktura tego grafu jest opisana w README.md
     """
-    # TODO: treść funkcji
-    pass
+    g = ig.Graph(directed=True)
+    v = g.add_vertex(is_root=True, unpaired_count_0=0, unpaired_count_1=0, unpaired_count_2=0)
+    opened_parentheses_count = 0
+
+    for c in parenthesized:
+        if c == ".":
+            v[f"unpaired_count_{v.outdegree()}"] = (v.attributes().get(f"unpaired_count_{v.outdegree()}") or 0) + 1
+        if c == "(":
+            opened_parentheses_count += 1
+            v_prev = v
+            v = g.add_vertex(is_root=False, unpaired_count_0=0, unpaired_count_1=0)
+            g.add_edge(v_prev, v)
+        if c == ")":
+            opened_parentheses_count -= 1
+            if (len(g.predecessors(v)) == 0):
```

Comment:
Nie wystarczy sprawdzić czy opened_parentheses_count < 0? Nie za duża różnica ale jeśli mamy dużo przypadków rozpatrywać to zawsze to przyspieszy jakoś

Reply:
Tak. Nawet po to wprowadziłem tę zmienną tylko zapomniałem jej tu użyć 🤦🤦‍♀️🤦‍♂️. Poprawię przy okazji. Zostaw tę rozmowę "unresolved" plis, żebym pamiętał żeby do tego wrócić.